src/battle/battle_command.c

```diff
@@ -5909,3 +5916,561 @@ static void *BattleScriptGetVarPointer(BattleSystem *bsys, BattleContext *ctx, i
+        if (Pokemon_TryLevelUp(mon)) {
+            // Only play the special level-up animation for an active battler
+            if (data->ctx->selectedMonIndex[expBattler] == slot) {
+                BattleController_EmitSetStatus2Effect(data->bsys, data->ctx, expBattler, 8);
+                ov12_0226399C(data->bsys, expBattler);
+            }
+
+            data->state = STATE_GET_EXP_WAIT_LEVEL_UP_EFFECT;
+        } else {
+            data->state = STATE_GET_EXP_CHECK_DONE;
+        }
+        break;
+
+    case STATE_GET_EXP_WAIT_LEVEL_UP_EFFECT:
+        if (Link_QueueNotEmpty(data->ctx)) {
+            TempStatsStruct stats = ov12_0226C354;
+            //TempDataLoader(ov12_0226C354, stats);
+            int level = GetMonData(mon, MON_DATA_LEVEL, NULL);
+            // Cache the stats from the previous level for later
+            data->ctx->unk_17C = AllocFromHeap(HEAP_ID_BATTLE, sizeof(StatStruct));
+            StatStruct *oldStats = data->ctx->unk_17C;
+            for (i = 0; i < 6; i++) {
+                oldStats->stats[i] = GetMonData(mon, stats.stats[i], NULL);
+            }
+
+            MonApplyFriendshipMod(mon, 0, BattleSystem_GetLocation(data->bsys));
+            ApplyMonMoodModifier(mon, 0);
+            CalcMonStats(mon);
+
+            if (data->ctx->selectedMonIndex[expBattler] == slot) {
+                BattleSystem_ReloadMonData(data->bsys,
+                    data->ctx,
+                    expBattler,
+                    data->ctx->selectedMonIndex[expBattler]);
+            }
+
+            data->ctx->levelUpMons |= MaskOfFlagNo(slot);
+            ov12_02263A1C(data->bsys, data->ctx, expBattler);
+
+            msg.id = 3; // "{0} grew to Lv. {1}!"
+            msg.tag = 17;
+            msg.param[0] = expBattler | (slot << 8);
+            msg.param[1] = level;
+            data->unk30[0] = ov12_0223C494(data->bsys, msgLoader, &msg, BattleSystem_GetTextFrameDelay(data->bsys));
+            data->state = STATE_GET_EXP_WAIT_LEVEL_UP_MESSAGE_PRINT;
+        }
+        break;
+
+    case STATE_GET_EXP_WAIT_LEVEL_UP_MESSAGE_PRINT:
+        if (TextPrinterCheckActive(data->unk30[0]) == 0) {
+            data->state = STATE_GET_EXP_LEVEL_UP_SUMMARY_LOAD_ICON;
+            data->unk30[2] = 0;
+        }
+        break;
+
+    case STATE_GET_EXP_LEVEL_UP_SUMMARY_LOAD_ICON:
+        // Load the Pokemon's Party icon if they are not the active battler
+        if (data->ctx->selectedMonIndex[expBattler] != slot) {
+            ov12_02248228(data->bsys, data, mon);
+        }
+
+        data->state = STATE_GET_EXP_LEVEL_UP_SUMMARY_INIT;
+        break;
+
+    case STATE_GET_EXP_LEVEL_UP_SUMMARY_INIT: {
+        BgConfig *bgConfig = BattleSystem_GetBgConfig(data->bsys);
+        Window *window = BattleSystem_GetWindow(data->bsys, 1);
+        PaletteData *palette = BattleSystem_GetPaletteData(data->bsys);
+
+        G2_SetBG0Priority(1 + 1); // this is the background + 1; could do with a constant
```

Comment:
Use a constant.

Reply:
We don't have constants for this + I don't know it well enough to add one, I'll just keep it at 2 for now